src/Orleans/Storage/OrleansRelationalExtensions.cs

```diff
@@ -0,0 +1,1506 @@
+                command.Parameters.Add(receivedMessagesCountParameter);
+
+                var connectionGatewayCountParameter = CreateConnectionGatewayCountParameter(command, clientMetrics.ConnectedGatewayCount, direction);
+                command.Parameters.Add(connectionGatewayCountParameter);
+            }).ConfigureAwait(continueOnCapturedContext: false);
+        }
+
+
+        /// <summary>
+        /// Inserts the given statistics counters to the Orleans database.
+        /// </summary>
+        /// <param name="storage">The storage to use.</param>
+        /// <param name="deploymentId">The deployment ID.</param>
+        /// <param name="hostName">The hostname.</param>
+        /// <param name="siloOrClientName">The silo or client name.</param>
+        /// <param name="id">The silo address or client ID.</param>
+        /// <param name="counters">The counters to be inserted.</param>        
+        public static async Task InsertStatisticsCountersAsync(this IRelationalStorage storage, string deploymentId, string hostName, string siloOrClientName, string id, IEnumerable<ICounter> counters)
+        {
+            //The query template from which to retrieve the set of columns that are being
+            //inserted.
+            var queryTemplate = OrleansRelationalConstants.GetConstant(storage.InvariantName, OrleansRelationalConstants.InsertOrleansStatisticsKey);
+
+            //Zero statistic values mean either that the system is not running or no updates. Such values are not inserted and pruned
+            //here so that no insert query or parameters are generated.
+            var countersList = counters.Where(i => !"0".Equals(i.IsValueDelta ? i.GetDeltaString() : i.GetValueString())).ToList();
+
+            //Note that the following is almost the same as RelationalStorageExtensions.ExecuteMultipleInsertIntoAsync
+            //the only difference being that some columns are skipped. Likely it would be beneficial to introduce
+            //a "skip list" to RelationalStorageExtensions.ExecuteMultipleInsertIntoAsync.
+
+            //The template contains an insert for online. The part after SELECT is copied
+            //out so that certain parameters can be multiplied by their count. Effectively
+            //this turns a query of type (transaction details vary by vendor)
+            //BEGIN TRANSACTINION; INSERT INTO [OrleansStatisticsTable] <columns> SELECT <variables>; COMMIT TRANSACTION;
+            //to BEGIN TRANSACTINION; INSERT INTO [OrleansStatisticsTable] <columns> SELECT <variables>; UNION ALL <variables> COMMIT TRANSACTION;
+            //where the UNION ALL is multiplied as many times as there are counters to insert.
+            int startFrom = queryTemplate.IndexOf("SELECT") + "SELECT".Length + 1; //This +1 is to have a space between SELECT and the first parameter name to not to have a SQL syntax error.
+            int lastSemicolon = queryTemplate.LastIndexOf(";");
```

Comment:
IndexOf and LastIndexOf are cultural specific string comparisons. 
I think you want to make them use StringComparison.Ordinal.

[Found by ReSharper scan]